api/src/main/java/com/datatorrent/api/StringCodec.java

```diff
@@ -377,4 +383,43 @@ public String toString(Class<? extends T> clazz)
     private static final long serialVersionUID = 201312082053L;
   }
 
+  public class JsonStringCodec<T> implements StringCodec<T>, Serializable
+  {
+    private static final long serialVersionUID = 2513932518264776006L;
+    Class<?> clazz;
+
+    public JsonStringCodec(Class<T> clazz)
+    {
+      this.clazz = clazz;
+    }
+
+    @Override
+    public T fromString(String string)
+    {
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ObjectReader reader = mapper.reader(clazz);
+        return reader.readValue(string);
+      } catch (IOException e) {
+        DTThrowable.wrapIfChecked(e);
+      }
+      return null;
+    }
+
+    @Override
+    public String toString(T pojo)
+    {
+      try {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationConfig.Feature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ObjectWriter writer = mapper.writer();
+        return writer.writeValueAsString(pojo);
+      } catch (IOException e) {
+        DTThrowable.wrapIfChecked(e);
```

Comment:
This is not specific to this PR but in general, should we consider using Throwables.propagate() instead?